src/build/env.py

```diff
@@ -182,9 +182,20 @@ def _create_isolated_env_venv(path):  # type: (str) -> Tuple[str, str]
     pip_distribution = next(iter(metadata.distributions(name='pip', path=[purelib])))
     pip_version = packaging.version.Version(pip_distribution.version)
 
-    # Currently upgrade if Pip 19.1+ not available, since Pip 19 is the first
-    # one to officially support PEP 517, and 19.1 supports manylinux1.
+    needs_pip_upgrade = False
     if pip_version < packaging.version.Version('19.1'):
+        # PEP-517 and manylinux1 was first implemented in 19.1
+        needs_pip_upgrade = True
+    elif platform.system() == 'Darwin':
+        # macOS 11+ name scheme change requires 20.3. Intel macOS 11.0 can be told to report 10.16 for backwards
+        # compatibility; but that also fixes earlier versions of pip so this is only needed for 11+.
+        if int(cast(Any, platform.mac_ver)()[0].split('.')[0]) >= 11:
```

Comment:
Let's split this up:

```suggestion
        # ``platform.mac_ver`` is untyped for Python 2 in the typeshed
        #   TODO: remove when upgrading mypy from 0.800
        platform_mac_ver = platform.mac_ver  # type: Any
        mac_major_ver = int(platform_mac_ver()[0].split('.')[0])
        if mac_major_ver >= 11:
```

Reply:
I disagree, so @FFY00 will need to decide.

Reply:
I would actually prefer to put a Python version check and `# type: ignore` as that allows us to easily identify the code to be removed when we drop Python 2 and still does the proper checks on Python 3 without any impact.

Reply:
See my comment above, I tried that first and MyPy doesn't seem to handle it correctly.

Reply:
Though maybe a blend of the two? Python version check and cast? (Also, this will go away in the next version of MyPy)

Reply:
Hum, what if you put the comment in the else rather than the assign? Anyway, even if that doesn't work, we can still only do the custom handling there on Python 2.

Reply:
Doesn't like that ignore at all. Tried a couple of places. I've put the cast in only for Python 2, and have a blend of the two versions above. ;)

Reply:
Thanks.